use soot API analysis java source code error

I use gradle to build soot to analyze the java source code, but the following error occurs every time. The versions I use are all 2.7.1, including **soot-infoflow: 2.7.1, soot-infoflow-android: 2.7.1, soot-infoflow-cmd: 2.7.1, soot-infoflow-summaries: 2.7.1**. As long as the source code contains ArrayList objects or print statements, errors will occur. How can I avoid this error? Can I only use bytecode files? What should I do if I can only use the source code?

`
Exception in thread "main" java.lang.Error: Error: Failed to load java.lang.CharSequence.
	at soot.JastAddJ.PathPart.getCompilationUnit(PathPart.java:109)
	at soot.JastAddJ.Program.getCompilationUnit(Program.java:873)
	at soot.JastAddJ.Program.getLibCompilationUnit_compute(Program.java:1517)
	at soot.JastAddJ.Program.getLibCompilationUnit(Program.java:1500)
	at soot.JastAddJ.Program.lookupLibType_compute(Program.java:1465)
	at soot.JastAddJ.Program.lookupLibType(Program.java:1433)
	at soot.JastAddJ.Program.lookupType_compute(Program.java:1412)
	at soot.JastAddJ.Program.lookupType(Program.java:1393)
	at soot.JastAddJ.Program.Define_TypeDecl_lookupType(Program.java:1981)
	at soot.JastAddJ.ASTNode.Define_TypeDecl_lookupType(ASTNode.java:1657)
	at soot.JastAddJ.ASTNode.Define_TypeDecl_lookupType(ASTNode.java:1657)
	at soot.JastAddJ.ASTNode.Define_TypeDecl_lookupType(ASTNode.java:1657)
	at soot.JastAddJ.ASTNode.Define_TypeDecl_lookupType(ASTNode.java:1657)
	at soot.JastAddJ.ASTNode.Define_TypeDecl_lookupType(ASTNode.java:1657)
	at soot.JastAddJ.Expr.lookupType(Expr.java:1186)
	at soot.JastAddJ.TypeAccess.decls_compute(TypeAccess.java:425)
	at soot.JastAddJ.TypeAccess.decls(TypeAccess.java:414)
	at soot.JastAddJ.TypeAccess.refined_TypeScopePropagation_TypeAccess_decl(TypeAccess.java:387)
	at soot.JastAddJ.TypeAccess.decl_compute(TypeAccess.java:460)
	at soot.JastAddJ.TypeAccess.decl(TypeAccess.java:452)
	at soot.JastAddJ.TypeAccess.type_compute(TypeAccess.java:590)
	at soot.JastAddJ.TypeAccess.type(TypeAccess.java:583)
	at soot.JastAddJ.ClassDecl.isCircular_compute(ClassDecl.java:1696)
	at soot.JastAddJ.ClassDecl.isCircular(ClassDecl.java:1673)
	at soot.JastAddJ.ClassDecl.superclass(ClassDecl.java:240)
	at soot.JastAddJ.ClassDecl.supertypeClassDecl(ClassDecl.java:1841)
	at soot.JastAddJ.ClassDecl.subtype_compute(ClassDecl.java:1832)
	at soot.JastAddJ.ClassDecl.subtype(ClassDecl.java:1817)
	at soot.JastAddJ.TypeVariable.supertypeClassDecl(TypeVariable.java:1414)
	at soot.JastAddJ.ClassDecl.subtype_compute(ClassDecl.java:1832)
	at soot.JastAddJ.ClassDecl.subtype(ClassDecl.java:1796)
	at soot.JastAddJ.ClassDecl.instanceOf_compute(ClassDecl.java:1555)
	at soot.JastAddJ.ClassDecl.instanceOf(ClassDecl.java:1548)
	at soot.JastAddJ.ParTypeAccess.typeCheck(ParTypeAccess.java:121)
	at soot.JastAddJ.ASTNode.collectErrors(ASTNode.java:1284)
	at soot.JastAddJ.ASTNode.collectErrors(ASTNode.java:1292)
	at soot.JastAddJ.ASTNode.collectErrors(ASTNode.java:1292)
	at soot.JastAddJ.ASTNode.collectErrors(ASTNode.java:1292)
	at soot.JastAddJ.ASTNode.collectErrors(ASTNode.java:1292)
	at soot.JastAddJ.ASTNode.collectErrors(ASTNode.java:1292)
	at soot.JastAddJ.ASTNode.collectErrors(ASTNode.java:1292)
	at soot.JastAddJ.ASTNode.collectErrors(ASTNode.java:1292)
	at soot.JastAddJ.ASTNode.collectErrors(ASTNode.java:1292)
	at soot.JastAddJ.ASTNode.collectErrors(ASTNode.java:1292)
	at soot.JastAddJ.CompilationUnit.errorCheck(CompilationUnit.java:178)
	at soot.JastAddInitialResolver.formAst(JastAddInitialResolver.java:67)
	at soot.JavaClassSource.resolve(JavaClassSource.java:66)
	at soot.SootResolver.bringToHierarchyUnchecked(SootResolver.java:240)
	at soot.SootResolver.bringToHierarchy(SootResolver.java:214)
	at soot.SootResolver.bringToSignatures(SootResolver.java:279)
	at soot.SootResolver.bringToBodies(SootResolver.java:319)
	at soot.SootResolver.processResolveWorklist(SootResolver.java:164)
	at soot.SootResolver.resolveClass(SootResolver.java:134)
	at soot.Scene.loadClass(Scene.java:942)
	at soot.Scene.loadClassAndSupport(Scene.java:927)
	at soot.Scene.loadNecessaryClasses(Scene.java:1753)
	at CallGraphTest.main(CallGraphTest.java:26)
Caused by: java.lang.ArrayIndexOutOfBoundsException: 20492
Caused by: java.lang.ArrayIndexOutOfBoundsException: 20492
	at soot.JastAddJ.BytecodeParser.parseThisClass(BytecodeParser.java:437)
	at soot.JastAddJ.BytecodeParser.parseTypeDecl(BytecodeParser.java:413)
	at soot.JastAddJ.BytecodeParser.parse(BytecodeParser.java:331)
	at soot.JastAddJ.BytecodeParser.read(BytecodeParser.java:27)
	at soot.JastAddJ.PathPart.getCompilationUnit(PathPart.java:87)
	... 56 more
`

--- a/src/main/generated/jastadd/soot/JastAddJ/BytecodeParser.java
+++ b/src/main/generated/jastadd/soot/JastAddJ/BytecodeParser.java
@@ -641,6 +641,15 @@ public class BytecodeParser extends java.lang.Object implements Flags, BytecodeR
     private static final int CONSTANT_Utf8 = 1;
 
 
+    private static final int CONSTANT_MethodHandle= 15;
+
+
+    private static final int CONSTANT_MethodType = 16;
+
+
+    private static final int CONSTANT_InvokeDynamic = 18;
+
+
 
     public void parseEntry(int i) {
       int tag = u1();
@@ -677,6 +686,15 @@ public class BytecodeParser extends java.lang.Object implements Flags, BytecodeR
           break;
         case CONSTANT_Utf8:
           constantPool[i] = new CONSTANT_Utf8_Info(this);
+          break;
+        case CONSTANT_MethodHandle:
+          constantPool[i] = new CONSTANT_MethodHandle_Info(this);
+          break;
+        case CONSTANT_MethodType:
+          constantPool[i] = new CONSTANT_MethodType_Info(this);
+          break;
+        case CONSTANT_InvokeDynamic:
+          constantPool[i] = new CONSTANT_InvokeDynamic_Info(this);
           break;
         default:
           println("Unknown entry: " + tag);

--- a/src/main/generated/jastadd/soot/JastAddJ/CONSTANT_InvokeDynamic_Info.java
+++ b/src/main/generated/jastadd/soot/JastAddJ/CONSTANT_InvokeDynamic_Info.java
@@ -1,0 +1,26 @@
+package soot.JastAddJ;
+
+/**
+ * @ast class
+ */
+public class CONSTANT_InvokeDynamic_Info extends CONSTANT_Info {
+
+    public int method_attr_index;
+
+
+    public int name_and_type_index;
+
+
+    public CONSTANT_InvokeDynamic_Info(BytecodeParser parser) {
+      super(parser);
+      method_attr_index = p.u2();
+      name_and_type_index = p.u2();
+    }
+
+
+    public String toString() {
+        return "InvokeDynamicInfo: " + method_attr_index + " " + name_and_type_index;
+    }
+
+
+}

--- a/src/main/generated/jastadd/soot/JastAddJ/CONSTANT_MethodHandle_Info.java
+++ b/src/main/generated/jastadd/soot/JastAddJ/CONSTANT_MethodHandle_Info.java
@@ -1,0 +1,26 @@
+package soot.JastAddJ;
+
+/**
+ * @ast class
+ */
+public class CONSTANT_MethodHandle_Info extends CONSTANT_Info {
+
+    public int reference_kind;
+
+
+    public int reference_index;
+
+
+    public CONSTANT_MethodHandle_Info(BytecodeParser parser) {
+      super(parser);
+      reference_kind = p.u1();
+      reference_index = p.u2();
+    }
+
+
+    public String toString() {
+        return "MethodHandleInfo: " + reference_kind + " " + reference_index;
+    }
+
+
+}

--- a/src/main/generated/jastadd/soot/JastAddJ/CONSTANT_MethodType_Info.java
+++ b/src/main/generated/jastadd/soot/JastAddJ/CONSTANT_MethodType_Info.java
@@ -1,0 +1,22 @@
+package soot.JastAddJ;
+
+/**
+ * @ast class
+ */
+public class CONSTANT_MethodType_Info extends CONSTANT_Info {
+
+    public int descriptor_index;
+
+
+    public CONSTANT_MethodType_Info(BytecodeParser parser) {
+      super(parser);
+      descriptor_index = p.u2();
+    }
+
+
+    public String toString() {
+        return "MethodTypeInfo: " + descriptor_index ;
+    }
+
+
+}

